tests/cases/compiler/declarationEmitCastReusesTypeNode4.ts

```diff
@@ -0,0 +1,45 @@
+// @strict: true
+// @strictNullChecks: true,false
+// @allowJs: true
+// @checkJs: true
+// @emitDeclarationOnly: true
+// @declaration: true
+// @noTypesAndSymbols: true
+// @filename: input.js
+/**
+ * @typedef {{ } & { name?: string }} P
+ */
+
+const something = /** @type {*} */(null);
+
+export let vLet = /** @type {P} */(something);
+export const vConst = /** @type {P} */(something);
+
+export function fn(p = /** @type {P} */(something)) {}
+
+/** @param {number} req */
+export function fnWithRequiredDefaultParam(p = /** @type {P} */(something), req) {}
+
+export class C {
+    field = /** @type {P} */(something);
+    /** @optional */ optField = /** @type {P} */(something); // not a thing
+    /** @readonly */ roFiled = /** @type {P} */(something);
+    method(p = /** @type {P} */(something)) {}
+    /** @param {number} req */
+    methodWithRequiredDefault(p = /** @type {P} */(something), req) {}
+
+    constructor(ctorField = /** @type {P} */(something)) {}

```

Comment:
This looks like a typo; should be `public ctorField =`..., right?

Reply:
This is javascript, so no - since parameter properties are a syntax error in JS files.

Reply:
(And there's no jsdoc equivalent)

Reply:
Oh, whoops, didn't notice that.